core/src/main/java/com/sequenceiq/cloudbreak/core/bootstrap/service/host/ClusterHostServiceRunner.java

```diff
@@ -662,12 +664,18 @@ public Map<String, SaltPillarProperties> createPillarWithClouderaManagerSettings
                                 "gov_cloud", govCluster(stackDto.getPlatformVariant()),
                                 "disable_auto_bundle_collection", disableAutoBundleCollection,
                                 "set_cdp_env", isVersionNewerOrEqualThanLimited(cmVersion, CLOUDERAMANAGER_VERSION_7_0_2),
+                                "cloud_provider_setup_supported", isCloudProviderSetupSupported(stackDto, cmVersion),
                                 "deterministic_uid_gid", isVersionNewerOrEqualThanLimited(cmVersion, CLOUDERAMANAGER_VERSION_7_2_1),
                                 "cloudera_scm_sudo_access", CMRepositoryVersionUtil.isSudoAccessNeededForHostCertRotation(clouderaManagerRepo)),
                         "mgmt_service_directories", serviceLocations.getAllVolumePath(),
                         "address", primaryGatewayConfig.getPrivateAddress()))));
     }
 
+    private boolean isCloudProviderSetupSupported(StackDto stackDto, String cmVersion) {
+        List<String> supportedCloudProviders = List.of(CloudPlatform.AWS.name(), CloudPlatform.GCP.name(), CloudPlatform.AZURE.name());
```

Comment:
this should be a constant

Reply:
what kind of failures did you experience? I'm not able to see how this could cause any issue

Reply:
This was the failure 

`Datalake creation failed. Data Lake creation failed on 'satya-dl4' cluster. Reason: Stack status: CREATE_FAILED, cluster status: CREATE_FAILED, reason: Failed: Orchestrator component went failed in 7 min(s), message: Salt execution went wrong:: ["Rendering SLS 'base:cloudera.agent' failed: Jinja variable 'dict object' has no attribute ''"]`

Reply:
this looks more like:
- missing pillar info from the host
- faulty salt state

and not related to changing this to static